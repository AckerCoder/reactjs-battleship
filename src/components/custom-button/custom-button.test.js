import React from "react";
import { shallow } from "enzyme";
import CustomButton from "./custom-button.component";

it("Expect to Match Snapshot", () => {
  expect(shallow(<CustomButton />)).toMatchSnapshot();
});

it("Expect to render CustomButton Component", () => {
  const mockChildren = "Mock Children";
  const mockLabel = "Mock Label";
  expect(
    shallow(
      <CustomButton label={mockLabel} main>
        {mockChildren}
      </CustomButton>
    ).length
  ).toEqual(1);
  expect(
    shallow(
      <CustomButton label={mockLabel} setting>
        {mockChildren}
      </CustomButton>
    ).length
  ).toEqual(1);
});
