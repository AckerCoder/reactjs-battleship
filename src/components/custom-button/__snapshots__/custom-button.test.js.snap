// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect to Match Snapshot 1`] = `
<div
  className="custom-button"
>
  <button
    className=" "
  />
</div>
`;
